react/SearchContext.js

```diff
@@ -62,10 +62,30 @@ const SearchContext = ({
     withFacets: includeFacets(mapField, queryField),
   }
 
+  const queryVariables = queryField ? customSearch : defaultSearch
+
+  const [facetQuery, facetMap] = zip(
+    queryVariables.query.split('/'),
+    queryVariables.map.split(',')
+  )
+    .filter(([_, map]) => map === 'c' || map === 'ft')
+    .reduce(
+      ([queryAcc, mapAcc], [query, map]) => [
+        [...queryAcc, query],
+        [...mapAcc, map],
+      ],
+      // first empty array is for query and the second
+      // for map
+      [[], []]
```

Comment:
couldn't you use an object here instead of arrays? so it would be mostly self-explanatory and you wouldn't need this comment